infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/AsyncStorageModeValidator.java

```diff
@@ -0,0 +1,158 @@
+import static org.eclipse.che.workspace.infrastructure.kubernetes.namespace.pvc.EphemeralWorkspaceUtility.isEphemeral;
+
+import com.google.common.base.Strings;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.eclipse.che.api.core.ValidationException;
+import org.eclipse.che.api.workspace.server.WorkspaceAttributeValidator;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Validates the workspace attributes before workspace creation and updating if async storage
+ * configure.
+ *
+ * <p>To be valid for async storage workspace MUST have attributes:
+ *
+ * <ul>
+ *   <li>{@link org.eclipse.che.api.workspace.shared.Constants#ASYNC_PERSIST_ATTRIBUTE} = 'true'
+ *   <li>{@link org.eclipse.che.api.workspace.shared.Constants#PERSIST_VOLUMES_ATTRIBUTE} = 'false'
+ * </ul>
+ *
+ * <p>If set only {@link org.eclipse.che.api.workspace.shared.Constants#ASYNC_PERSIST_ATTRIBUTE} =
+ * 'true', {@link ValidationException} is thrown.
+ *
+ * <p>If system is configured with other value of properties than below {@link ValidationException},
+ * is thrown.
+ *
+ * <ul>
+ *   <li>che.infra.kubernetes.namespace.default=<username>-che
+ *   <li>che.infra.kubernetes.namespace.allow_user_defined=false
+ *   <li>che.infra.kubernetes.pvc.strategy=common
+ *   <li>che.limits.user.workspaces.run.count=1
+ * </ul>
+ */
+public class AsyncStorageModeValidator implements WorkspaceAttributeValidator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AsyncStorageModeValidator.class);
+
+  private final String pvcStrategy;
+  private final boolean allowUserDefinedNamespaces;
+  private final String defaultNamespaceName;
+  private final int runtimesPerUser;
+
+  @Inject
+  public AsyncStorageModeValidator(
+      @Named("che.infra.kubernetes.pvc.strategy") String pvcStrategy,
+      @Named("che.infra.kubernetes.namespace.allow_user_defined")
+          boolean allowUserDefinedNamespaces,
+      @Nullable @Named("che.infra.kubernetes.namespace.default") String defaultNamespaceName,
+      @Named("che.limits.user.workspaces.run.count") int runtimesPerUser) {
+
+    this.pvcStrategy = pvcStrategy;
+    this.allowUserDefinedNamespaces = allowUserDefinedNamespaces;
+    this.defaultNamespaceName = defaultNamespaceName;
+    this.runtimesPerUser = runtimesPerUser;
+  }
+
+  @Override
+  public void validate(Map<String, String> attributes) throws ValidationException {
+    if (parseBoolean(attributes.get(ASYNC_PERSIST_ATTRIBUTE))) {
+      isEphemeralAttributeValidation(attributes);
+      pvcStrategyValidation();
+      alowUserDefinedNamespaceValidation();
+      nameSpaceStrategyValidation();
+      runtimesPerUserValidation();
+    }
+  }
+
+  @Override
+  public void validateUpdate(Map<String, String> existing, Map<String, String> update)
+      throws ValidationException {
+    if (parseBoolean(update.get(ASYNC_PERSIST_ATTRIBUTE))) {
```

Comment:
NPE if the update doesn't contain the attribute at all

Reply:
`parseBoolean` will parse `null` to `false` and `Map.get` should return `null` in case of missing key.